src/main/java/com/bettercloud/vault/rest/Rest.java

```diff
@@ -169,10 +169,12 @@ public Rest parameter(final String name, final String value) throws RestExceptio
      * @throws RestException If any error occurs, or unexpected response received from Vault
      */
     public Rest header(final String name, final String value) throws RestException {
-        try {
-            this.headers.put(URLEncoder.encode(name, "UTF-8"), URLEncoder.encode(value, "UTF-8"));
-        } catch (UnsupportedEncodingException e) {
-            throw new RestException(e);
+        if (value != null) {
+            try {
+                this.headers.put(URLEncoder.encode(name, "UTF-8"), URLEncoder.encode(value, "UTF-8"));
```

Comment:
URL Encoding will break functionality when X-Vault-Namespace value value is a child namespace.  "infosec/devops" gets changed to "infsoec%2Fdevops" .